apps/web/__tests__/support/pageObjects/CartPageObject.ts

```diff
@@ -73,19 +71,20 @@ export class CartPageObject {
     return this;
   }
 
-  orderSummayAfterCouponApplyed(discount: string, total: string) {
+  orderSummayAfterCouponApplyed() {
     cy.getByTestId('coupon-label').should('be.visible');
-    cy.getByTestId('coupon-value').contains(discount);
-    cy.getByTestId('total').contains(total);
+    cy.getByTestId('coupon-value').invoke('text').should('have.length.gt', 0);
+    cy.getByTestId('total').invoke('text').should('have.length.gt', 0);
+    return this;
   }
 
   removeCoupon() {
     cy.getByTestId('couponRemove').click();
     return this;
   }
 
-  orderSummayAfterCouponRemoved(total: string) {
+  orderSummayAfterCouponRemoved() {
     cy.getByTestId('coupon-label').should('not.exist');
-    cy.getByTestId('total').contains(total);
+    cy.getByTestId('total').invoke('text').should('have.length.gt', 0);
```

Comment:
**question**: What do we gain from checking the text itself here? Compared to something like this:

```
orderSummayAfterCouponApplyed() {
    cy.getByTestId('coupon-label').should('be.visible');
    cy.getByTestId('coupon-value').should('be.visible');
  }

  removeCoupon() {
    cy.getByTestId('couponRemove').click();
    return this;
  }

  orderSummayAfterCouponRemoved() {
    cy.getByTestId('coupon-label').should('not.exist');
    cy.getByTestId('coupon-value').should('not.exist');
```

**suggestion**: I don't think this is necessarily the appropriate place to check the `total` element either. That one's part of the regular order summary, regardless of whether we apply a coupon or not. We might want to introduce a separate method instead, e.g. `cart.hasOrderSummary`, that checks for the order summary's structure as a whole.